openedx/stanford/djangoapps/course_utils/management/commands/course_grep.py

```diff
@@ -0,0 +1,131 @@
+"""
+Search component data for specific text
+"""
+from __future__ import unicode_literals
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from opaque_keys.edx.keys import CourseKey
+from xmodule.capa_module import CapaDescriptor
+from xmodule.html_module import HtmlBlock
+from xmodule.modulestore.django import modulestore
+
+_supported_component_types = [
+    'capa',
+    'html',
+]
+
+
+class Command(BaseCommand):
+    """
+    Search component data for specific text
+    """
+    help = __doc__
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-t',
+            '--type',
+            action='append',
+            choices=_supported_component_types,
+            help=(
+                'Limit search to specific component types. '
+                'By default, search all component types. '
+            ),
+        )
+        parser.add_argument(
+            '-c',
+            '--course',
+            action='append',
+            help=(
+                'Limit search to a specific course. '
+                'By default, search all courses. '
+            ),
+        )
+        parser.add_argument(
+            'search_terms',
+            help=(
+                'Specify one or more search terms, '
+                'case-sensitive. '
+            ),
+            nargs='*',
+        )
+
+    def handle(self, *args, **options):
+        """
+        Execute the command
+        """
+        terms = options['search_terms'] or ['', ]
+        component_types = options.get('type', _supported_component_types)
+        course_ids = options.get('course')
+        results = search(terms, component_types, course_ids)
+        for course, component in results:
+            output = format_output(course, component)
+            print(output)
+
+
+def format_output(course, component):
+    base = settings.CMS_BASE
+    url = "{base}/container/{module_id}".format(
+        base=base,
+        course_id=course.id,
+        module_id=component.location,
+    )
+    data = component.data
+    data = data.split('\n')
+    data = ''.join(data)
+    output = "{course_id}	{url}	{data}".format(
```

Comment:
Output is now a tab-delimited list of `course_id`, `url`, and `data`.

`data` has any newlines stripped, meaning results can be parsed like:

```sh
# Get all URLs
./manage.py cms --settings=devstack course_grep 2>/dev/null | cut -f2
```